src/main/scala/mesosphere/util/state/MesosLeaderInfo.scala

```diff
@@ -0,0 +1,30 @@
+package mesosphere.util.state
+
+/**
+  * Utility class for keeping track of a Mesos Leader
+  */
+trait MesosLeaderInfo {
+
+  def currentLeaderUrl: Option[String]
+
+  def load(host: String, port: Int): Unit
+}
+
+case class ConstMesosLeaderInfo(currentLeaderUrl: Option[String]) extends MesosLeaderInfo {
+
+  override def load(host: String, port: Int): Unit = {}
+}
+
+object MutableMesosLeaderInfo extends MesosLeaderInfo {
```

Comment:
You asked me directly whether you use a global variable for this or not. For some reason, I misinterpreted the question. Otherwise I would have said: No, please don't! ;) Sorry about that.

So please make `MutableMesosLeaderInfo` a class instead of an object.